app/page.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import { fetchWeather } from "./store/slices/weather";
+import { useDispatch, useSelector } from "react-redux";
+import { useState, useEffect } from "react";
+import Chart from "./components/chart";
+import type { weatherState } from "./types/weatherState";
+import { indexData } from "./types/indexData";
+import type { cityProcessedData } from "./types/cityProcessedData";
+
+type weatherPropsType = {
```

Comment:
It's convention to `TitleCase` type names. Also to prefer `interface` over `type` where posible